frontend-react/src/App.tsx

```diff
@@ -64,6 +64,12 @@ const App = () => {
         navigate(toRelativeUrl(originalUri, window.location.origin));
     };
 
+    // watch the session broadcast channel to handle login/logout on other tabs
+    const bc = new BroadcastChannel("session");
```

Comment:
Ah, this'll run on every re-render of the App component, which happens on navigation, context change, and auth state change -- so we'll be creating new instances of BroadcastChannel and adding a listener each time.  Let's use a `useRef` and a `useEffect` so it'll only do that once:

```jsx
// only create one BroadcastChannel instance for this component
const broadcastChannelRef = useRef(new BroadcastChannel("session"));

// only bind the message listener on mount
useEffect(() => {
  broadcastChannelRef.current.onmessage = () => {
    OKTA_AUTH.authStateManager.updateAuthState();
  };

  // unbind the message listener on unmount
  return () => broadcastChannelRef.current.close();
}, []);
```

Reply:
👍